test/e2e/framework/gpu_util.go

```diff
@@ -69,7 +70,16 @@ func NVIDIADevicePlugin(ns string) *v1.Pod {
 
 func GetGPUDevicePluginImage() string {
 	ds, err := DsFromManifest(GPUDevicePluginDSYAML)
-	if err != nil || ds == nil || len(ds.Spec.Template.Spec.Containers) < 1 {
+	if err != nil {
+		glog.Errorf("Failed to parse the device plugin image: %v", err)
+		return ""
+	}
+	if ds == nil {
+		glog.Errorf("Failed to parse the device plugin image: the extracted DaemonSet is nil")
+		return ""
+	}
+	if len(ds.Spec.Template.Spec.Containers) < 1 {
```

Comment:
Any specific reason why no containers needs to be checked? Wouldn't DsFromManifest return that in the err field?

Reply:
Actually I'm not sure about the specific reason, maybe pre-checking would return error ASAP.